packages/repl/src/index.js

```diff
@@ -278,9 +281,20 @@ export function createRepl({ isVerbose, onAssert, onLog, onOk }) {
     const importObject = {
       _internalInstanceOptions: internalInstanceOptions
     };
-    const module = createCompiledModule(moduleNode);
 
-    return new Instance(module, importObject);
+    if (enableTypeChecking === true) {
+      denormalizeTypeReferences(moduleNode);
+
+      const typeErrors = typeCheck(t.program([moduleNode]));
+
+      if (typeErrors.length > 0) {
+        throw new Error("type mismatch");
+      }
+    }
+
+    const compiledModule = createCompiledModule(moduleNode);
```

Comment:
See the implementation of `createCompiledModule`: https://github.com/xtuc/webassemblyjs/blob/master/packages/webassemblyjs/src/compiler/compile/module.js#L40-L43. 

It calls the root validation function which includes the new typechecker.

Why do we need to run the typechecker before again? It seems that there's a bug.